assets/js/blocks/featured-product/block.js

```diff
@@ -103,17 +105,28 @@ class FeaturedProduct extends Component {
 		const { productId } = this.props.attributes;
 		if ( ! productId ) {
 			// We've removed the selected product, or no product is selected yet.
-			this.setState( { product: false, loaded: true } );
+			this.setState( { product: false, loaded: true, error: false } );
 			return;
 		}
 		apiFetch( {
 			path: `/wc/blocks/products/${ productId }`,
 		} )
 			.then( ( product ) => {
-				this.setState( { product, loaded: true } );
+				this.setState( { product, loaded: true, error: false } );
 			} )
-			.catch( () => {
-				this.setState( { product: false, loaded: true } );
+			.catch( ( apiError ) => {
+				const error = {
+					retry: this.debouncedGetProduct,
+				};
+
+				if ( isObject( apiError ) ) {
+					error.message = __( 'The following error was returned from the API', 'woo-gutenberg-products-block' ) + ': <br/><code>' + apiError.message + '</code>';
```

Comment:
Besides the comment above, do you think we should wrap `apiError.message` with `escapeHTML` just to play it safe?